ckanext/cfpb_extrafields/plugin.py

```diff
@@ -35,8 +36,71 @@ def popup_usage_restrictions():
     return "Enter instructions for what users can and cannot do with the data."
 
 class ExampleIDatasetFormPlugin(p.SingletonPlugin, tk.DefaultDatasetForm):
-    p.implements(p.IDatasetForm)
-    p.implements(p.IConfigurer)
+
+    p.implements(p.IResourceController)
+    def before_create(self, context, resource):
+        return
+    def after_create(self, context, resource):
+        # must have changed['format'] when you created the resource
+        tk.redirect_to(controller='package',action='resource_edit',
+                       resource_id=resource['id'],id=resource['package_id'])
+
+    def _control_datadict(self, resource):
+        ''' delete and rebuild datadict'''
+        record = resource.get('datadict','')
+        if record:
+            resource.pop('datadict', None)
+            record = eval(record) 
+            ds.delete_datadict(resource['id'])
+            ds.create_datadict(resource['id'],record,'') 
```

Comment:
What happens here if create_datadict returns failure?  We shouldn't let it silently fail